spec/lib/artifact.spec.js

```diff
@@ -0,0 +1,280 @@
+"use strict";
+
+describe("lib/artifact", () => {
+    var Artifact, authTokenMock, bluebird, content, downloadLocation, fs, instance, logger, requestMock, requestOptionsMock, requestPromiseMock, responseHandlerMock, responseMock, uriMock;
+
+    authTokenMock = "abcd1234";
+    bluebird = require("bluebird");
+    downloadLocation = "example/download/path";
+    fs = require("fs");
+    logger = require("../../lib/logger")();
+
+    /**
+     * A mock Metadata class.
+     */
+    class MetadataMock {
```

Comment:
Moved metadataMock into  its own file for now. If the rest of this task comes in under time I may make the TestContainer.